packages/confirmation-modal/src/ConfirmationModal.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import { css } from '@leafygreen-ui/emotion';
+import Button from '@leafygreen-ui/button';
+import Modal, { Footer } from '@leafygreen-ui/modal';
+import { uiColors } from '@leafygreen-ui/palette';
+import TextInput from '@leafygreen-ui/text-input';
+
+const titleStyle = css`
+  color: ${uiColors.gray.dark2};
+  font-size: 24px;
+  font-weight: bold;
+  line-height: 25px;
+
+  margin-bottom: 10px;
+`;
+
+const baseModalStyle = css`
+  width: 600px;
+  padding: initial;
+`;
+
+const contentStyle = css`
+  color: ${uiColors.gray.dark1};
+  font-family: Akzidenz, ‘Helvetica Neue’, Helvetica, Arial, sans-serif;
+  font-size: 16px;
+  letter-spacing: 0;
+  line-height: 24px;
+
+  padding: 36px;
+`;
+
+const textEntryInputStyle = css`
+  width: 300px;
+  margin-top: 14px;
+`;
+
+interface ConfirmationModalProps {
+  title: string;
+  children: React.ReactNode;
+  open?: boolean;
+  onClose?: (confirmed: boolean) => void;
+  className?: string;
+  buttonText: string;
+  dangerous?: boolean;
+  requiredInputText?: string;
+}
+
+const ConfirmationModal = ({
+  children,
+  title,
+  requiredInputText,
+  buttonText,
+  dangerous = false,
```

Comment:
We should probably use the variant pattern that we've established in other components. Perhaps `variant={'danger' | 'default'}` could work fine here.